### PR TITLE
Performance: Added fast_double_parser to improve double conversion time

### PR DESCRIPTION
**Pull request by filipecosta90**
This PR improves string to double conversion by leveraging  a C Wrapper for the C++ Eisel-Lemire ParseFloat algorithm.
This optimizes primarily our write workflows and we've measured an improvement of ~=8% on the achievable ops/sec. 

Bellow you can find the top-down tree of cpu usage that made us focus on this optimization. 

## Top-Down tree of hotspots by CPU utilization for TS.MADD....
Please notice the 16.7% line, which we're optimizing on this PR. 

Function Stack | CPU Time: Total | CPU Time: Self | Function (Full) | Source File
-- | -- | -- | -- | --
add | 96.10% | 0.122s | add | module.c
RM_OpenKey | 27.70% | 0.066s | RM_OpenKey | module.c
internalAdd | 24.80% | 0.404s | internalAdd | module.c
**RM_StringToDouble** | **16.70%** | 0.038s | **RM_StringToDouble** | module.c
RM_CloseKey | 8.30% | 0.114s | RM_CloseKey | module.c
RM_StringToLongLong | 6.80% | 0.082s | RM_StringToLongLong | module.c
zfree | 4.20% | 0.262s | zfree | zmalloc.c
ParseDuplicatePolicy | 2.40% | 0.184s | ParseDuplicatePolicy | query_language.c
RM_ModuleTypeGetType | 1.90% | 0.236s | RM_ModuleTypeGetType | module.c
__GI_ | 1.50% | 0.212s | float __GI_(long, int, bool, char) | malloc.c

## Validating improvements with a tsbs dataset
on a quick look, using a single shard to test we move from 1.4M datapoints/sec ( 140K ops/sec of TS.MADD with 10 points ) to 1.53M datapoints.

According to Eisel-Lemire ParseFloat algorithm numbers, we should see a 4X improvement on the string to double, which means we would move from 17% CPU time on conversion to 0.25*17%=4.25% on best case scenario. I'm seeing an real improvement of ~8-9% on ingestion (meaning we're at 2X improvement ) :
```
###########################
Standalone on master branch
###########################
Summary:
loaded 120960000 metrics in 85.206sec with 1 workers (mean rate 1419615.30 metrics/sec)
loaded 12097000 rows in 85.206sec with 1 workers (mean rate 141973.27 rows/sec)

###########################
Standalone on fast.doubleparser branch
###########################
Summary:
2021/03/23 20:08:17 finished benchmark
loaded 120960000 metrics in 78.765sec with 1 workers (mean rate 1535699.71 metrics/sec)
loaded 12097000 rows in 78.765sec with 1 workers (mean rate 153582.67 rows/sec)
```

### CODE REVIEW

**Review by danni-m — APPROVED** [review]
Looks great to me

**Review by danni-m — APPROVED** [review]

**Review by danni-m — DISMISSED** [review]

File: src/Makefile [hunk 1]
```diff
@@ -66,6 +69,7 @@ CC=gcc
 CC_FLAGS = \
 	-I$(SDK_DIR) \
 	-I$(ROOT)/deps \
+	-I$(SRCDIR)/fast_double_parser_c \
```
Comment (rafie) [inline src/Makefile:72]: Not needed (with the change in `module.c`).

**Review by rafie — COMMENTED** [review]
lVgtm.
Few comments.

File: src/module.c [hunk 2]
```diff
@@ -12,6 +12,7 @@
 #include "common.h"
 #include "compaction.h"
 #include "config.h"
+#include "fast_double_parser_c.h"
```
Comment (rafie) [inline src/module.c:15]: `#include "fast_double_parser_c/fast_double_parser_c.h"`

**Review by rafie — APPROVED** [review]

**Review by danni-m — APPROVED** [review]

**Pull request merged**